ipet/parsing/Solver.py

```diff
@@ -0,0 +1,305 @@
+
+import re
+import os
+from ipet.misc import misc
+from ipet.parsing.Key import Key
+from .StatisticReader import StatisticReader
+
+class Solver(StatisticReader):
+    
+    DEFAULT=None
+    DEFAULT_SOLVERID="defaultSolver"
+    
+    def __init__(self, 
```

Comment:
The constructor could be simplified by using groups and regular expressions as follows, I exemplify this for the primal bound. Currently, a pattern, and an index are used to distinguish the primal bound. It would be much simpler to use the following, combined expression:

```
inputstr = "Primal Bound       : +3.02000000000000e+02 (1 solutions)"
import re
primalboundexpression = re.compile("Primal Bound       : (\S+)")
print primalboundexpression.match(inputstr).groups()

>>> ('+3.02000000000000e+02',)

```

The design would be the following: The constructor receives patterns (strings) or None, and compiles expressions such as the above.

The method extractPrimalBound() would be changed to
```
try:
     pb = float(self.primalboundexpression.groups()[0])
except:
     pb = None
```
In a nutshell:

- Let's base parsing on simpler patterns such as the above to simplify the constructor, otherwise we drown in argument checking hell.

- Subclasses should also directly specify the expressions as static class member like so:

```
class SCIPSolver(Solver):

     primalbound_expression = re.compile("Primal Bound       : (\S+)")
     dualbound_expression = re.compile(...)
     ... remaining expressions...

    def __init__(self, primalbound_pattern=None):
         if primalbound_pattern is not None:
              self.primalbound_expression = re.compile(primalbound_pattern)
         ...
```